operator/scripts/update-kcc-manifest/main.go

```diff
@@ -268,18 +270,79 @@ func main() {
 		log.Fatalf("error updating file %v", stableFilePath)
 	}
 
-	staleManifestDir := path.Join(operatorSrcRoot, "channels", "packages", "configconnector", currentVersion.Version)
-	log.Printf("removing stale manifest %v", staleManifestDir)
-	if err := os.RemoveAll(staleManifestDir); err != nil {
-		log.Fatal(fmt.Errorf("error deleting dir %v: %w", staleManifestDir, err))
+	channelDir := path.Join(operatorSrcRoot, "channels", "packages", "configconnector")
+	if err := dropStalePackages(channelDir); err != nil {
+		log.Fatalf("drop stale packages: %s", err)
 	}
-	staleManifestDir = path.Join(operatorSrcRoot, "autopilot-channels", "packages", "configconnector", currentVersion.Version)
-	log.Printf("removing stale manifest %v", staleManifestDir)
-	if err := os.RemoveAll(staleManifestDir); err != nil {
-		log.Fatal(fmt.Errorf("error deleting dir %v: %w", staleManifestDir, err))
+	autoPilotChannelDir := path.Join(operatorSrcRoot, "autopilot-channels", "packages", "configconnector")
+	if err := dropStalePackages(autoPilotChannelDir); err != nil {
+		log.Fatalf("drop stale packages: %s", err)
 	}
 }
 
+func dropStalePackages(pacakgesPath string) error {
+	// packagesPath := path.Join(operatorSrcRoot, "channels", "packages", "configconnector")
+
+	dirEntries, _ := os.ReadDir(pacakgesPath)
+	versionMap := map[int][]int{}
+	minorVersions := []int{}
+	totalVersions := []string{}
+	// Build map, key is the SemVer2 Minor, value is a slice of the SemVer2 Patches
+	for _, entry := range dirEntries {
+		if entry.IsDir() {
+			version, err := semver.ParseTolerant(entry.Name())
+			if err != nil {
+				log.Printf("skipping unknown package version %q under %s", entry.Name(), pacakgesPath)
+				continue
+			}
+			totalVersions = append(totalVersions, entry.Name())
+			minor := int(version.Minor)
+			patch := int(version.Patch)
+			if curPatches, ok := versionMap[minor]; !ok {
+				versionMap[minor] = []int{patch}
+				minorVersions = append(minorVersions, minor)
+			} else {
+				curPatches = append(curPatches, patch)
+				versionMap[minor] = curPatches
+			}
+		} else {
+			log.Printf("found unknown file %s under %s\n", entry.Name(), pacakgesPath)
+		}
+	}
+	supportedVersions := []semver.Version{}
```

Comment:
Nit: it might easier just to build this list from the directories, sort it, then delete all but the last N.